server/src/main/java/org/apache/iotdb/db/metadata/LocalConfigNode.java

```diff
@@ -759,23 +717,29 @@ public void dropSchemaTemplate(DropTemplatePlan plan) throws MetadataException {
   public synchronized void setSchemaTemplate(SetTemplatePlan plan) throws MetadataException {
     PartialPath path = new PartialPath(plan.getPrefixPath());
     try {
-      getBelongedSchemaRegionWithAutoCreate(path).setSchemaTemplate(plan);
+      schemaEngine
+          .getSchemaRegion(getBelongedSchemaRegionIdWithAutoCreate(path))
+          .setSchemaTemplate(plan);
     } catch (StorageGroupAlreadySetException e) {
       throw new MetadataException("Template should not be set above storageGroup");
     }
   }
 
   public synchronized void unsetSchemaTemplate(UnsetTemplatePlan plan) throws MetadataException {
+    PartialPath path = new PartialPath(plan.getPrefixPath());
     try {
-      getBelongedSchemaRegion(new PartialPath(plan.getPrefixPath())).unsetSchemaTemplate(plan);
+      schemaEngine
+          .getSchemaRegion(getBelongedSchemaRegionIdWithAutoCreate(path))
+          .unsetSchemaTemplate(plan);
     } catch (StorageGroupNotSetException e) {
       throw new PathNotExistException(plan.getPrefixPath());
     }
   }
 
   public void setUsingSchemaTemplate(ActivateTemplatePlan plan) throws MetadataException {
+    PartialPath path = plan.getPrefixPath();
     try {
-      getBelongedSchemaRegion(plan.getPrefixPath()).setUsingSchemaTemplate(plan);
+      schemaEngine.getSchemaRegion(getBelongedSchemaRegionId(path)).setUsingSchemaTemplate(plan);
```

Comment:
```SetUsingTemplate``` may trigger auto create mnode, which means there may be auto create of schemaRegion.